packages/admin/admin/src/form/Field.tsx

```diff
@@ -51,9 +51,12 @@ export function Field<FieldValue = any, FieldElement extends HTMLElement = HTMLE
     const validateError = required ? (validate ? composeValidators(requiredValidator, validate) : requiredValidator) : validate;
 
     const finalFormContext = useFinalFormContext();
-    const shouldShowError = passedShouldShowError || finalFormContext.shouldShowFieldError;
-    const shouldShowWarning = passedShouldShowWarning || finalFormContext.shouldShowFieldWarning;
-    const shouldScrollToField = passedShouldScrollTo || finalFormContext.shouldScrollToField;
+    const shouldShowError =
+        passedShouldShowError ?? ((fieldMeta: FieldMetaState<FieldValue>) => finalFormContext.shouldShowFieldError({ fieldMeta }));
+    const shouldShowWarning =
+        passedShouldShowWarning ?? ((fieldMeta: FieldMetaState<FieldValue>) => finalFormContext.shouldShowFieldWarning({ fieldMeta }));
+    const shouldScrollToField =
+        passedShouldScrollTo ?? ((fieldMeta: FieldMetaState<FieldValue>) => finalFormContext.shouldScrollToField({ fieldMeta }));
```

Comment:
Since the `FinalFormContext` is also public API, I couldn't change its typing either. So I wrapped the method calls.